### DISCUSSION TITLE
SYCL runtime_errors, CUDA BE

### DISCUSSION THREAD

**Discussion by dosarudaniel** [intel/llvm#4392] [Q&A]
Hi,

This is a follow up of this [discussion](https://github.com/intel/llvm/discussions/3627).

By following @steffenlarsen's instructions (Many thanks, btw!) on this [thread](https://github.com/intel/llvm/discussions/3627) from github I managed to obtain the converted code which compiles (using these [instructions](https://github.com/dosarudaniel/oneAdePT#readme)) and links (following the same [receipt](https://github.com/dosarudaniel/oneAdePT/blob/687eb98038cd55bda643e9450e84dca15aa8d65e/Linking_example9.1.txt) as the one [suggested](https://github.com/intel/llvm/discussions/3627#discussioncomment-1136268) on github) with success now, but when I run example9.1 binary file I came across this error:
### 1. 
```
terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  No kernel named  was found -46 (CL_INVALID_KERNEL_NAME)
```
The error occurred at [this line](https://github.com/dosarudaniel/oneAdePT/blob/687eb98038cd55bda643e9450e84dca15aa8d65e/examples/Example9.1/example9.dp.cpp#L273), as the [gdb backtrace](https://imgur.com/a/igKGQgn) showed.
I tried [this](https://github.com/intel/llvm/issues/1427#issuecomment-668780304) possible solution, but it didn't helped me.

Trying to debug the above error, I change the code into [something simpler](https://github.com/dosarudaniel/oneAdePT/blob/0a12187cf2e7071c8bb2eb8f28e8ed5823ec7edb/examples/Example9.1/example9.dp.cpp#L279-L300) taken from this [tutorial](https://github.com/codeplaysoftware/syclacademy/blob/c38eacddd6e1719a44b96932a066c198bee6d9bb/Code_Exercises/Exercise_03_Scalar_Add/solution.cpp) (and linked using the same [receipt](https://github.com/dosarudaniel/oneAdePT/blob/687eb98038cd55bda643e9450e84dca15aa8d65e/Linking_example9.1.txt)), but I got this error at runtime:
### 2. 
```
terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  Invalid device program image: size is zero -30 (CL_INVALID_VALUE)
```
The error occurred at [this line](https://github.com/dosarudaniel/oneAdePT/blob/0a12187cf2e7071c8bb2eb8f28e8ed5823ec7edb/examples/Example9.1/example9.dp.cpp#L284), as the [gdb backtrace 2](https://imgur.com/a/j2P7TLB) showed.
    
  
**_Do you have any tips on how to approach these issues ?_**

   
I am using this version of the clang compiler: e8f2166fdd09ce3f6a15bcb7b5783cbbd149c0f7 
The code from this  [tutorial](https://github.com/codeplaysoftware/syclacademy/blob/c38eacddd6e1719a44b96932a066c198bee6d9bb/Code_Exercises/Exercise_03_Scalar_Add/solution.cpp)  worked just fine outside my project following the tutorial's building instructions
I tried [this](https://github.com/intel/llvm/issues/1427#issuecomment-668780304) for the first issue, but did not helped me.

Thanks,
Daniel

**Comment by steffenlarsen** [comment]
Happy to hear my suggestion got you a bit further in your work. I have been trying to reproduce the issue you mention, but I have had little luck due to various dependencies. Do you per chance have a smaller reproducer?
